### PR TITLE
Refs #35058 -- Made OGRGeomType aware of additional WKB geometry types.

### PR DESCRIPTION
**Pull request by smithdc1**
This commit increases OGRGeomType's knowledge of WKB types. This allows for improved error messages when Django doesn't yet have a corresponding class to wrap a given type.

ogr_geometry types: [Link](https://github.com/OSGeo/gdal/blob/62c1489feab9312d0468abde0ad13d7b35ce27f7/ogr/ogr_core.h#L415)

Enum types for the "2.5D" types (for example `POINT Z`). There's some docs here. https://gdal.org/development/rfc/rfc49_curve_geometries.html. So even though the code used by GDAL is 1001 for `POINT Z` user code still sees `-2147483647`

> It has been considered if it would worth to modify the enumeration values of the existing 2.5D geometries (wkbPoint25D, etc...) to conform with the WKB codes of ISO SQL/MM Part 3 / SFA 1.2.1, but there was not a clear advantage in doing so, with respect to the impact on existing users of OGR API.

I suspect I've missed something trying to cover such a long list of values.

### CODE REVIEW

**Review by claudep — APPROVED** [review]
Nice improvement!

File: tests/gis_tests/gdal_tests/test_geom.py [hunk 1]
```diff
@@ -632,3 +632,82 @@ def test_empty(self):
...
+            ("MultiPoint", 4, True),
+            ("Multilinestring", 5, True),
+            ("MultiPolygon", 6, True),
+            ("GeometryCollection", 7, True),
+            ("CircularString", 8, False),
+            ("CompoundCurve", 9, False),
+            ("CurvePolygon", 10, False),
+            ("MultiCurve", 11, False),
+            ("MultiSurface", 12, False),
+            # 13 (Curve) and 14 (Surface) are absract types.
```
Comment (felixxm) [inline tests/gis_tests/gdal_tests/test_geom.py:650]:
```suggestion
            # 13 (Curve) and 14 (Surface) are abstract types.
```

File: django/contrib/gis/gdal/geometries.py [hunk 2]
```diff
@@ -123,7 +123,10 @@ def __init__(self, geom_input, srs=None):
             self.srs = srs
 
         # Setting the class depending upon the OGR Geometry Type
-        self.__class__ = GEO_CLASSES[self.geom_type.num]
+        geo_class = GEO_CLASSES.get(self.geom_type.num, None)
+        if geo_class is None:
```
Comment (felixxm) [inline django/contrib/gis/gdal/geometries.py:127]:
```suggestion
        if (geo_class := GEO_CLASSES.get(self.geom_type.num, None)) is None:
```

File: django/contrib/gis/geometry.py [hunk 3]
```diff
@@ -10,7 +10,8 @@
     r"^(SRID=(?P<srid>\-?[0-9]+);)?"
     r"(?P<wkt>"
     r"(?P<type>POINT|LINESTRING|LINEARRING|POLYGON|MULTIPOINT|"
-    r"MULTILINESTRING|MULTIPOLYGON|GEOMETRYCOLLECTION)"
+    r"MULTILINESTRING|MULTIPOLYGON|GEOMETRYCOLLECTION|CIRCULARSTRING|SURFACE|TIN|"
+    r"POLYHEDRALSURFACE|TRIANGLE|CURVE|MULTISURFACE|MULTICURVE|COMPOUNDCURVE)"
```
Comment (felixxm) [inline django/contrib/gis/geometry.py:14]: Should we also add `CURVEPOLYGON`?

**Comment by felixxm** [comment]
@smithdc1 Thanks :+1: I left small comments.

File: django/contrib/gis/geometry.py (hunk 3 above)
Comment (smithdc1) [reply django/contrib/gis/geometry.py:14]: Yes. That looks right.

I'm away for a ~week now so will push edits when I get back. (Happy too if you want to push it over the line in the meantime)

Comment (felixxm) [reply django/contrib/gis/geometry.py]: Pushed edits.

**Comment by smithdc1** [comment]
Thanks all!

**Pull request merged**